app.py

```diff
@@ -93,12 +100,33 @@ def postIndex():
         handleUpload(f, js=False)
     return redirect(url_for('getIndex'))
 
+@app.route('/upload', methods=['POST'])
+def postIndexAPI():
+    """
+    This will handle uploads to the API, returning a JSON consisting
+    of original file names and their corresponding uploaded URLs
+    """
+    flash(Markup("Uploading"))
+    uploaded = request.files.getlist("file[]")
+    files = []
+    urls = []
+    for f in uploaded:
+        files.append(str(f.filename))
+        v = handleUpload(f, js=False, api=True)
+        urls.append(v)
+    res = "{"
+    for url, f in zip(urls, files):
+        res += f + ":" + url + ","
```

Comment:
Wouldn't it be a better idea to make a list of dictionaries and serialize it to JSON? This way it wouldn't even work since f and url won't be received as strings.